### PR TITLE
Feature/quadratic solve

### LINKED ISSUE #432: Solve function for quadratic solvers
I think your solution was very elegant. We might do it with the `quadratic` branch as well?

_Originally posted by @marcocuturi in https://github.com/ott-jax/ott/issues/401#issuecomment-1712443280_

### PR DESCRIPTION
**Pull request by michalk8**
closes #432

### CODE REVIEW

File: src/ott/solvers/quadratic/_solve.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      reduces to a plain Gromov-Wasserstein problem :cite:`peyre:16`.
+    fused_penalty: Multiplier of the linear term in fused Gromov-Wasserstein,
+      i.e. ``problem = purely quadratic + fused_penalty * linear problem``.
+    a: The first marginal. If :obj:`None`, it will be uniform.
+    b: The second marginal. If :obj:`None`, it will be uniform.
+    tau_a: If :math:`< 1`, defines how much unbalanced the problem is
+      on the first marginal.
+    tau_b: If :math:`< 1`, defines how much unbalanced the problem is
+      on the second marginal.
+    loss: A 2-tuple of 2-tuples of Callable. The first tuple is the linear
```
Comment (marcocuturi) [inline src/ott/solvers/quadratic/_solve.py:56]: maybe this part is a bit cryptic... and does not match the typing above. I am guessing the original was no clearer :) 

Maybe just reference to the `quadratic_costs.GWLoss` docs and try to be clearer there?

File: src/ott/solvers/quadratic/_solve.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      part of the loss. The second one is the quadratic part (quad1, quad2).
+      By default, the loss is set as the 4 functions representing the squared
+      Euclidean loss, and this property is taken advantage of in subsequent
+      computations. Alternatively, KL loss can be specified in no less optimized
+      way. Only used when ``rank = -1``.
+    gw_unbalanced_correction: Whether the unbalanced version of
+      :cite:`sejourne:21` is used. Otherwise, ``tau_a`` and ``tau_b``
+      only affect the inner Sinkhorn loop. Only used when ``rank = -1``.
```
Comment (marcocuturi) [inline src/ott/solvers/quadratic/_solve.py:64]: "only affect the inner Sinkhorn loop" -> "only affect the resolution of the linearization of the GW problem in the inner loop"

**Review by marcocuturi — COMMENTED** [review]
this is starting to look good! thanks a lot Michal

File: src/ott/solvers/quadratic/_solve.py (hunk 1 above)
Comment (michalk8) [reply src/ott/solvers/quadratic/_solve.py]: Updated, thanks for the suggestion!

File: docs/tutorials/notebooks/gromov_wasserstein_multiomics.ipynb [hunk 3]
```diff
@@ -359,11 +349,10 @@
    ]
   },
   {
-   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We provide the Average FOSCTTM to align `X` (chromatinaccessibility domain) to `Y` (gene expression domain) for each implementation:"
+    "We provide the Average FOSCTTM to align `X` (chromatin accessibility domain) to `Y` (gene expression domain) for each implementation:"
```
Comment (marcocuturi) [inline docs/tutorials/notebooks/gromov_wasserstein_multiomics.ipynb:355]: thanks for the catch!

File: src/ott/problems/quadratic/quadratic_costs.py [hunk 4]
```diff
@@ -24,23 +24,54 @@ class Loss(NamedTuple):  # noqa: D101
   is_linear: bool
 
 
-class GWLoss(NamedTuple):  # noqa: D101
+class GWLoss(NamedTuple):
```
Comment (marcocuturi) [inline src/ott/problems/quadratic/quadratic_costs.py:27]: thanks

**Review by marcocuturi — APPROVED** [review]
LGTM, thanks!

**Pull request merged**